File: AddFraction.java
package algebra;

import java.util.ArrayList;
import java.util.List;

public class AddFraction {

    protected final List<Integer> numerators;
    protected final List<Integer> denominators;
    protected List<Integer> new_numerators;
    protected int lcm;
    protected int answer = 0;

    public AddFraction(List<Integer> num, List<Integer> denom) {
        numerators = num;
        denominators = denom;
        new_numerators = new ArrayList<>();
    }

    protected void canonizeFraction() {
        LCM l_c_m = new LCM(denominators);
        lcm = l_c_m.getLCM();

        for (int i = 0; i < denominators.size(); i++) {
            new_numerators.add(lcm / denominators.get(i) * numerators.get(i));
        }
    }

    public int[] doAdd() {
        canonizeFraction();

        for (int n : new_numerators) {
            answer += n;
        }
        return new int[]{answer, lcm};
    }
}

File: AddFraction.cs
using System.Collections.Generic;

namespace Algebra_CS
{
    class AddFraction
    {
        protected List<int> numerators = new List<int>();
        protected List<int> denominators = new List<int>();
        protected List<int> new_numerators = new List<int>();
        protected int lcm;
        protected int answer = 0;

        public AddFraction(List<int> num, List<int> denom)
        {
            numerators = num;
            denominators = denom;
        }

        protected void canonizeFraction()
        {
            // STEP 1:
            LCM l_c_m = new LCM(denominators);
            lcm = l_c_m.getLCM();

            // STEP 2:
            for (int i = 0; i < denominators.Count; i++)
            {
                new_numerators.Add(lcm / denominators[i] * numerators[i]);
            }
        }

        public int[] doAdd()
        {
            canonizeFraction();

            // STEP 3:
            foreach (int n in new_numerators)
            {
                answer += n;
            }
            return new int[] { answer, lcm };
        }
    }
}
